server/route.go

```diff
@@ -410,6 +416,20 @@ func (c *client) processRouteInfo(info *Info) {
 		return
 	}
 
+	// Detect if we have a mismatch of cluster names.
+	if info.Cluster != "" && info.Cluster != clusterName {
+		c.mu.Unlock()
+		// If we are dynamic we may update our cluster name.
+		if c.srv.getOpts().Cluster.Name == "" && strings.Compare(clusterName, info.Cluster) < 0 {
+			s.setClusterName(info.Cluster)
+			s.removeAllRoutesExcept(c)
```

Comment:
This kind of scare me: we set our cluster name, then remove all but this route, but you have to think of this server accepting other routes at the same time. There can be some delay in connecting due to this.. hoping that it will ultimately settle.

Reply:
Added isClusterNameDynamic()

Reply:
The one test makes sure it always settle, creates random number of servers up to 20 but I ran the test at 100 as well for count 100.